src/renderers/shaders/ShaderChunk/lights_pars.glsl

```diff
@@ -107,6 +107,46 @@ uniform vec3 ambientLightColor;
 
 #if defined( USE_ENVMAP ) && defined( PHYSICAL )
 
+
+	vec3 getDiffuseLightProbeIndirectLightColor( /*const in SpecularLightProbe specularLightProbe,*/ const in GeometricContext geometry, const in int maxMIPLevel ) { 
+
+		#ifdef DOUBLE_SIDED
+
+			float flipNormal = ( float( gl_FrontFacing ) * 2.0 - 1.0 );
+
+		#else
+
+			float flipNormal = 1.0;
+
+		#endif
+
+		vec3 worldNormal = inverseTransformDirection( geometry.normal, viewMatrix );
+
+		#ifdef ENVMAP_TYPE_CUBE
+
+			vec3 queryVec = flipNormal * vec3( flipEnvMap * worldNormal.x, worldNormal.yz );
+
+			#if defined( TEXTURE_CUBE_LOD_EXT )				
+
+				vec4 envMapColor = textureCubeLodEXT( envMap, queryVec, float( maxMIPLevel ) );
+
+			#else
+
+				vec4 envMapColor = textureCube( envMap, queryVec, float( maxMIPLevel ) );
+
```

Comment:
BTW @tschw I'm done with changes to the lighting model.  I had some other refactors but they are not that important and I think they can wait as I have to get back to Clara.io stuff for the time being.

So if you had time and inclination to do your suggested cubeToUV2 change, that is likely the most important thing for improving the quality. :)  But it is up to you and the other people on this project.